Update operating mode upon network disagreement.

mtrippled opened the pull request.

ripplelabs-jenkins commented:
## Jenkins Build Summary

Built from [this commit](https://github.com/ripple/rippled/commit/82dd46b88671d3dbdbe3934e14c9e0d0d5fa651c)

Built at __20190723 - 01:50:05__

### Test Results

Build Type | Log | Result | Status
---------- | --- | ------ | ------
msvc.Debug | [logfile](http://ci.ops.ripple.com/job/rippled-dev/job/PR-3006/1//artifact/msvc.Debug.txt) | 1178 cases, 0 failed, t: 570s | PASS :white_check_mark: 
gcc.Release <br/>-Dassert=ON, <br/>MANUAL_TESTS=true | [logfile](http://ci.ops.ripple.com/job/rippled-dev/job/PR-3006/1//artifact/gcc.Release_Dassert_ON_MANUAL_TESTS_true.txt) | 915 cases, 0 failed, t: 4m58s | PASS :white_check_mark: 
docs, <br/>TARGET=docs | [logfile](http://ci.ops.ripple.com/job/rippled-dev/job/PR-3006/1//artifact/docs_TARGET_docs.txt) | 1 cases, 0 failed, t: 0m1s | PASS :white_check_mark: 
gcc.Debug <br/>-Dcoverage=ON, <br/>TARGET=coverage_report, <br/>SKIP_TESTS=true | [logfile](http://ci.ops.ripple.com/job/rippled-dev/job/PR-3006/1//artifact/gcc.Debug_Dcoverage_ON_TARGET_coverage_report_SKIP_TESTS_true.txt) | 1178 cases, 0 failed, t: 16m50s | PASS :white_check_mark: 
clang.Debug | [logfile](http://ci.ops.ripple.com/job/rippled-dev/job/PR-3006/1//artifact/clang.Debug.txt) | 1178 cases, 0 failed, t: 2m53s | PASS :white_check_mark: 
msvc.Debug, <br/>NINJA_BUILD=true | [logfile](http://ci.ops.ripple.com/job/rippled-dev/job/PR-3006/1//artifact/msvc.Debug_NINJA_BUILD_true.txt) | 1178 cases, 0 failed, t: 599s | PASS :white_check_mark: 
clang.Debug <br/>-Dunity=OFF | [logfile](http://ci.ops.ripple.com/job/rippled-dev/job/PR-3006/1//artifact/clang.Debug_Dunity_OFF.txt) | 1177 cases, 0 failed, t: 3m51s | PASS :white_check_mark: 
gcc.Debug | [logfile](http://ci.ops.ripple.com/job/rippled-dev/job/PR-3006/1//artifact/gcc.Debug.txt) | 1178 cases, 0 failed, t: 2m53s | PASS :white_check_mark: 
clang.Release <br/>-Dassert=ON | [logfile](http://ci.ops.ripple.com/job/rippled-dev/job/PR-3006/1//artifact/clang.Release_Dassert_ON.txt) | 1178 cases, 0 failed, t: 5m8s | PASS :white_check_mark: 
gcc.Release <br/>-Dassert=ON | [logfile](http://ci.ops.ripple.com/job/rippled-dev/job/PR-3006/1//artifact/gcc.Release_Dassert_ON.txt) | 1178 cases, 0 failed, t: 4m57s | PASS :white_check_mark: 
gcc.Debug <br/>-Dunity=OFF | [logfile](http://ci.ops.ripple.com/job/rippled-dev/job/PR-3006/1//artifact/gcc.Debug_Dunity_OFF.txt) | 1177 cases, 0 failed, t: 11m44s | PASS :white_check_mark: 
msvc.Debug <br/>-Dunity=OFF | [logfile](http://ci.ops.ripple.com/job/rippled-dev/job/PR-3006/1//artifact/msvc.Debug_Dunity_OFF.txt) | 1177 cases, 0 failed, t: 1107s | PASS :white_check_mark: 
gcc.Debug <br/>-Dstatic=OFF | [logfile](http://ci.ops.ripple.com/job/rippled-dev/job/PR-3006/1//artifact/gcc.Debug_Dstatic_OFF.txt) | 1178 cases, 0 failed, t: 2m54s | PASS :white_check_mark: 
gcc.Debug <br/>-Dstatic=OFF -DBUILD_SHARED_LIBS=ON | [logfile](http://ci.ops.ripple.com/job/rippled-dev/job/PR-3006/1//artifact/gcc.Debug_Dstatic_OFF_DBUILD_SHARED_LIBS_ON.txt) | 1178 cases, 0 failed, t: 2m56s | PASS :white_check_mark: 
msvc.Release | [logfile](http://ci.ops.ripple.com/job/rippled-dev/job/PR-3006/1//artifact/msvc.Release.txt) | 1178 cases, 0 failed, t: 408s | PASS :white_check_mark: 
gcc.Debug, <br/>NINJA_BUILD=true | [logfile](http://ci.ops.ripple.com/job/rippled-dev/job/PR-3006/1//artifact/gcc.Debug_NINJA_BUILD_true.txt) | 1178 cases, 0 failed, t: 2m46s | PASS :white_check_mark: 
clang.Debug <br/>-Dunity=OFF -Dsan=address, <br/>PARALLEL_TESTS=false, <br/>DEBUGGER=false | [logfile](http://ci.ops.ripple.com/job/rippled-dev/job/PR-3006/1//artifact/clang.Debug_Dunity_OFF_Dsan_address_PARALLEL_TESTS_false_DEBUGGER_false.txt) | 1177 cases, 0 failed, t: 7m39s | PASS :white_check_mark: 
clang.Debug <br/>-Dunity=OFF -Dsan=undefined, <br/>PARALLEL_TESTS=false | [logfile](http://ci.ops.ripple.com/job/rippled-dev/job/PR-3006/1//artifact/clang.Debug_Dunity_OFF_Dsan_undefined_PARALLEL_TESTS_false.txt) | 1177 cases, 0 failed, t: 4m16s | PASS :white_check_mark:

nbougalis commented:
@mtrippled please rebase this

mtrippled commented on src/ripple/app/misc/NetworkOPs.cpp:
```diff
@@ -620,11 +623,16 @@ static std::array<char const*, 5> const stateNames {{
-static_assert (NetworkOPs::omSYNCING == 2, "");
-static_assert (NetworkOPs::omTRACKING == 3, "");
-static_assert (NetworkOPs::omFULL == 4, "");
+static_assert (static_cast<std::size_t>(
```
fixed

mtrippled commented on src/ripple/app/misc/NetworkOPs.h:
```diff
@@ -76,7 +76,7 @@ class NetworkOPs
     using clock_type = beast::abstract_clock <std::chrono::steady_clock>;
 
-    enum OperatingMode
+    enum class OperatingMode
```
fixed

mtrippled commented:
In addition to the review comments, I fixed a race condition that I introduced in the original commit. There was a brief window in which the OperatingMode causing online_delete to abend could have changed by the time the log message was written.
